google_guest_agent/cfg/cfg.go

```diff
@@ -283,9 +291,9 @@ func defaultDataSources(extraDefaults []byte) []interface{} {
 
 	return append(res, []interface{}{
 		[]byte(defaultConfig),
-		configFile,
 		configFile + ".distro",
 		configFile + ".template",
+		configFile,
```

Comment:
Why are you changing the configuration precedence?

Reply:
Well, in this case I'd love if you could make a PR only for that. And yes, unit test is very much appreciated. :-)